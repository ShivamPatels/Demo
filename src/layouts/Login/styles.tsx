import { StyleSheet } from "react-native";
import { screenWidth } from "../../utils/dimensions";

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: "#fff",
        alignItems: "center",
        paddingTop: 80,
    },
    txtField: {
        marginTop: 20,
        width: screenWidth - 60,
    },
    signIn: {
        marginTop: 30,
        width: screenWidth - 40,
    },
    loginTitle: { fontWeight: "bold", fontSize: 20 },
});
export default styles;
