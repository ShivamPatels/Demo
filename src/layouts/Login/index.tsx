import React, { useState } from "react";
import { View, Text, ScrollView, Keyboard } from "react-native";
import { useForm, Controller } from "react-hook-form";

import TextField from "../../components/TextField";
import ErrorMessage from "../../utils/ErrorMessage";
import { REGEX } from "../../utils/validation";
import TextButton from "../../components/TextButton";
import colors from "../../utils/colors";
import Routes from "../../navigation/Route";
import LoadingIndicator from "../../components/LoadingIndicator";
import styles from "./styles";
import { postRequest } from "./../../api";
import Urls from "../../api/urls";

type Props = {
    navigation: any;
};

const Login = ({ navigation }: Props) => {
    const {
        control,
        handleSubmit,
        formState: { errors, isValid },
    } = useForm({ mode: "onChange" });

    const [isLoading, setIsLoading] = useState(false);

    const onSignInPress = (data: any) => {
        setIsLoading(true);

        Keyboard.dismiss();

        postRequest(Urls.LOGIN, {
            email: data.email,
            password: data.password,
        }).then(
            (response) => {
                setIsLoading(false);
                navigation.navigate(Routes.Detail);
            },
            (error) => {
                setIsLoading(false);
                if (
                    error &&
                    error.response &&
                    error.response.data &&
                    error.response.data.error
                ) {
                    alert(error.response.data.error);
                }
            }
        );
    };

    return (
        <View style={{ flex: 1 }}>
            <ScrollView contentContainerStyle={styles.container}>
                <Text style={styles.loginTitle}>LOGIN</Text>

                <Controller
                    control={control}
                    render={({ field: { onChange, value } }) => (
                        <TextField
                            mainStyle={styles.txtField}
                            placeholder="Email address"
                            title="Email"
                            value={value}
                            onChangeText={onChange}
                            errorMessage={errors?.email?.message}
                            isHideShow={false}
                        />
                    )}
                    name="email"
                    rules={{
                        required: {
                            value: true,
                            message: ErrorMessage.EMAIL_EMPTY,
                        },
                        pattern: {
                            value: REGEX.EMAIL,
                            message: ErrorMessage.EMAIL_INVALID,
                        },
                    }}
                />

                <Controller
                    control={control}
                    render={({ field: { onChange, value } }) => (
                        <TextField
                            mainStyle={styles.txtField}
                            title="Password"
                            placeholder="Password"
                            secureTextEntry={true}
                            value={value}
                            onChangeText={onChange}
                            errorMessage={errors?.password?.message}
                            isHideShow={true}
                        />
                    )}
                    name="password"
                    rules={{
                        required: {
                            value: true,
                            message: ErrorMessage.PASSWORD_EMPTY,
                        },
                    }}
                />

                <TextButton
                    style={[
                        styles.signIn,
                        {
                            backgroundColor: !isValid
                                ? colors.BG_BUTTON
                                : colors.BLACK,
                        },
                    ]}
                    disabled={!isValid}
                    titleStyle={{
                        color: !isValid ? colors.TXT_BUTTON : colors.WHITE,
                    }}
                    title={"Sign in"}
                    onPress={handleSubmit(onSignInPress)}
                />
            </ScrollView>
            {isLoading && <LoadingIndicator />}
        </View>
    );
};

export default Login;
