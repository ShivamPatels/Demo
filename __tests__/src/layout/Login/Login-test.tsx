/**
 * @format
 */

import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import Login from "../../../../src/layouts/Login";

it("renders correctly", () => {
    const tree = renderer.create(<Login navigation={null} />).toJSON();
    expect(tree).toMatchSnapshot();
});
