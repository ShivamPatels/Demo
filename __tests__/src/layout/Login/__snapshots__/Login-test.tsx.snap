// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
        "paddingTop": 80,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        LOGIN
      </Text>
      <View
        style={
          Array [
            Object {
              "marginTop": 20,
              "width": 690,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#000000",
              "fontWeight": "400",
              "marginVertical": 10,
            }
          }
        >
          Email
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#F2F2F7",
              "borderColor": "transparent",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": 48,
              "paddingHorizontal": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <TextInput
              autoCapitalize="none"
              autoCorrect={false}
              multiline={false}
              onChangeText={[Function]}
              placeholder="Email address"
              placeholderTextColor="rgba(60, 60, 67, 0.3)"
              style={
                Object {
                  "color": "#060518",
                  "flex": 1,
                  "fontSize": 15,
                  "fontWeight": "normal",
                  "height": 48,
                  "padding": 0,
                }
              }
            />
          </View>
        </View>
        <View />
      </View>
      <View
        style={
          Array [
            Object {
              "marginTop": 20,
              "width": 690,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#000000",
              "fontWeight": "400",
              "marginVertical": 10,
            }
          }
        >
          Password
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#F2F2F7",
              "borderColor": "transparent",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": 48,
              "paddingHorizontal": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <TextInput
              autoCapitalize="none"
              autoCorrect={false}
              multiline={false}
              onChangeText={[Function]}
              placeholder="Password"
              placeholderTextColor="rgba(60, 60, 67, 0.3)"
              secureTextEntry={true}
              style={
                Object {
                  "color": "#060518",
                  "flex": 1,
                  "fontSize": 15,
                  "fontWeight": "normal",
                  "height": 48,
                  "padding": 0,
                }
              }
            />
          </View>
        </View>
        <View />
      </View>
      <View
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#A8A8A8",
            "borderRadius": 40,
            "justifyContent": "center",
            "marginTop": 30,
            "opacity": 1,
            "width": 710,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#898A8D",
                "fontSize": 17,
                "fontWeight": "700",
                "paddingVertical": 12,
              },
              Object {
                "color": "#898A8D",
              },
            ]
          }
        >
           
          Sign in
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
